src/d3/scatterplot.d3.ts

```diff
@@ -10,46 +10,110 @@ export interface ScatterPlotStyle {
+    this.dataTextContainer.className = 'dataTxt';
+    container.appendChild(this.dataPointContainer);
+    container.appendChild(this.dataTextContainer);
+    if (this.data.length > 0){
+      this.generatePts();
+      this.generateText();
+      this.setColor('blue');
+    }
     this.createSky('gray');
     this.createGridPlane();
   }
-
-  private scaleTime(date: Date): number{
-    const startTime = this.data[0].x;
-    const endTime = this.data[this.data.length - 1].x;
-    this.timeScale = d3.scaleTime().domain([startTime, endTime]).rangeRound([0, this.TIME_MAX]);
-    return this.timeScale(date);
+  private scalePosition(){
+    let maxXValue = this.data[0].x;
+    let maxYValue = this.data[0].y;
+    let maxZValue = this.data[0].z;
+    for (const pt of this.data){
+      if (pt.x > maxXValue){
+        maxXValue = pt.x;
+      }
+      if (pt.y > maxYValue){
+        maxYValue = pt.y;
+      }
+      if (pt.z > maxZValue){
+        maxZValue = pt.z;
+      }
+    }
+    // scale positions based on largest value found in xyz to absVal(maxGridDimension)
+    this.xScale = d3.scaleLinear().domain([0, maxXValue]).range([0, this.GRID_BOUND]);
+    this.yScale = d3.scaleLinear().domain([0, maxYValue]).range([0, this.GRID_BOUND]);
+    this.zScale = d3.scaleLinear().domain([0, maxZValue]).range([0, this.GRID_BOUND]);
   }
   private generatePts() {
+    this.scalePosition();
      // enter identifies any DOM elements to be added when # array elements doesn't match
-    d3.select(this.container).selectAll(this.shape).data(this.data).enter().append(this.shape);
+    d3.select(this.dataPointContainer).selectAll(this.shape).data(this.data).enter().append(this.shape);
     // d is data at index, i within
     // select all shapes within given container
-    d3.select(this.container).selectAll(this.shape).attr('radius', .5).attr('position', (d, i) => {
-      const x = (d as VRScatterPoint).x;
-      const y = (d as VRScatterPoint).y;
-      const z = (d as VRScatterPoint).z;
+    d3.select(this.dataPointContainer).selectAll(this.shape).attr('radius', this.DATA_PT_RADIUS).attr('position', (d, i) => {
+      const x = this.xScale((d as VRScatterPoint).x);
+      const y = this.yScale((d as VRScatterPoint).y);
+      const z = this.zScale((d as VRScatterPoint).z);
       return `${x} ${y} ${z}`;
     });
   }
+  private generateText(){
+       // enter identifies any DOM elements to be added when # array elements doesn't match
+       d3.select(this.dataTextContainer).selectAll('a-entity').data(this.data).enter().append('a-entity');
+       d3.select(this.dataTextContainer).selectAll('a-plane').data(this.data).enter().append('a-plane')
+       .attr('position', (d, i) => {
+        const x = this.xScale((d as VRScatterPoint).x);
+        const y = this.yScale((d as VRScatterPoint).y);
+        const z = this.zScale((d as VRScatterPoint).z);
+        return `${x - 6 * this.DATA_PT_RADIUS} ${y + 2 * this.DATA_PT_RADIUS} ${z}`;
```

Comment:
Add comment explaining constants.

Reply:
since we are changing to hovering in front of camera, there will no longer be a need to add the padding in each dimension

now the user will just see a card in front of them based on what data point they hover on